src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GMMVariantTrain.java

```diff
@@ -0,0 +1,471 @@
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.Hidden;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hdf5.HDF5File;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.engine.MultiVariantWalker;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.clustering.BayesianGaussianMixtureModelPosterior;
+import org.broadinstitute.hellbender.utils.clustering.BayesianGaussianMixtureModeller;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+import org.broadinstitute.hellbender.utils.variant.VcfUtils;
+import picard.cmdline.programgroups.VariantFilteringProgramGroup;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * TODO
+ */
+@CommandLineProgramProperties(
+        summary = "Build a recalibration model to score variant quality for filtering purposes",
+        oneLineSummary = "Build a recalibration model to score variant quality for filtering purposes",
+        programGroup = VariantFilteringProgramGroup.class
+)
+@DocumentedFeature
+public class GMMVariantTrain extends MultiVariantWalker {
+
+    @ArgumentCollection
+    final private GMMVariantTrainArgumentCollection GMMVTAC = new GMMVariantTrainArgumentCollection();
+
+    /////////////////////////////
+    // Inputs
+    /////////////////////////////
+
+    /**
+     * Any set of VCF files to use as lists of training, truth, or known sites.
+     * Training - The program builds the Gaussian mixture model using input variants that overlap with these training sites.
+     * Truth - The program uses these truth sites to determine where to set the cutoff in VQSLOD sensitivity.
```

Comment:
Ended up going with the "calibration" label as suggested. I also made labeled resources optional so that the extract tool can be used for general purposes. In the event that no labeled resources are provided, we can still use the tool to write annotations for a reservoir-downsampled selection of sites to HDF5. Furthermore, we previously required that the special "training" and "calibration" labels were provided via resources; I've relaxed this as well.